api/src/main/java/io/opencensus/trace/MessageEvent.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016-17, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.Timestamp;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A class that represents a message event. It requires a {@link Type type} and a message id that
```

Comment:
this javadoc needs help, especially as it replaces network event. Instead of mentioning the field names, how about what this is used for? For example, messaging is like pubsub? What use cases does this help with? If the enum values are to be added later, what might they be? Is this application layer, network layer, transport layer? Just some things to help jog what this is for.

Reply:
Thanks! I will update that later.